### PR TITLE
[Reflection] Use actual type of element if constructor of custom attribute typed argument takes an array of objects.

### PR DESCRIPTION
**Pull request by MaximLipnin**
This PR adds several unit tests. The fix is in https://github.com/mono/corert/pull/27.

Fixes https://github.com/mono/mono/issues/10951

### CODE REVIEW

File: mcs/class/corlib/Test/System.Reflection/CustomAttributeDataTest.cs [hunk 1]
```diff
@@ -102,6 +139,77 @@ public string MethodWithReturnValueDecoratedWithMarshalAs ()
...
+			Assert.AreEqual (0, arr [0].Value, $"{methodName}#5");
+			Assert.AreEqual (typeof (int), arr [0].Value.GetType (), $"{methodName}#6");
+			
+			Assert.AreEqual (typeof (Levels), arr [1].ArgumentType, $"{methodName}#7");
+			Assert.IsTrue (arr [1].ArgumentType.GetTypeInfo().IsEnum, $"{methodName}#8");
+			Assert.AreEqual (1, arr [1].Value, $"{methodName}#9");
+			Assert.AreEqual (typeof (int), arr [1].Value.GetType (), $"{methodName}#10");
+		}
+
+		private void CheckObjectArrayParams ()
```
Comment (marek-safar) [inline mcs/class/corlib/Test/System.Reflection/CustomAttributeDataTest.cs]: Should be just normal [Test] method

File: mcs/class/corlib/Test/System.Reflection/CustomAttributeDataTest.cs [hunk 2]
```diff
@@ -102,6 +139,77 @@ public string MethodWithReturnValueDecoratedWithMarshalAs ()
 			Assert.AreEqual (typeof (byte), arr [1].ArgumentType);
 			Assert.AreEqual (2, arr [1].Value);
 		}
+
+		[Test]
+		// https://github.com/mono/mono/issues/10951
+		public void ObjectArrays () 
+		{
+			CheckObjectArrayParam (nameof (MethodDecoratedWithAttribute3));
```
Comment (marek-safar) [inline mcs/class/corlib/Test/System.Reflection/CustomAttributeDataTest.cs]: It's fine to duplicate the test code

**Comment by marek-safar** [comment]
@monojenkins backport 2018-10

**Comment by MaximLipnin** [comment]
Seems that another issue should be resolved here.

**Comment by marek-safar** [comment]
There is a lot of failing tests

**Comment by marek-safar** [comment]
@monojenkins backport 2018-10

**Comment by MaximLipnin** [comment]
Thanks for review!

**Pull request merged**